Can not build in MSVC

It seemed CMake is not working in oniguruma with MSVC, but use make_win64.bat works.

I tested the last updates on the windows branch. I am using `Visual Studio 2015`, I got this error:
```
:\User\Downloads\rust-onig>cargo build
  Compiling onig_sys v61.1.0 (file:///D:/User/Downloads/rust-onig/onig_sys)
rror: failed to run custom build command for `onig_sys v61.1.0 (file:///D:/User/Downloads/rust-onig/onig_sys)`
rocess didn't exit successfully: `D:\User\Downloads\rust-onig\target\debug\build\onig_sys-6d16105efd8e8e03\build-script-build` (exit code: 101)
-- stderr
hread 'main' panicked at 'Build error:
TDERR:
icrosoft (R) Program Maintenance Utility Version 14.00.23506.0
opyright (C) Microsoft Corporation.  All rights reserved.

MAKE : fatal error U1077: 'F:\VisualStudio2015\VC\bin\cl.EXE' : return code '0x2'
top.

TDOUT:
:\User\Downloads\rust-onig\onig_sys\oniguruma>cd src

:\User\Downloads\rust-onig\onig_sys\oniguruma\src>copy config.h.win64 config.h
       1 file(s) copied.

:\User\Downloads\rust-onig\onig_sys\oniguruma\src>nmake -f Makefile.windows
       cl -O2 -nologo /W3  -DHAVE_CONFIG_H -DNOT_RUBY -DEXPORT /I. /I.. /Foreggnu.obj /c reggnu.c
eggnu.c
:\user\downloads\rust-onig\onig_sys\oniguruma\src\regint.h(143): fatal error C1083: Cannot open include file: 'stdlib.h': No such file or directory
, onig_sys\build.rs:57
ote: Run with `RUST_BACKTRACE=1` for a backtrace.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,9 @@ rust:
 - beta
 - nightly
 
+branches:
+  only:
+    - master
 env:
   - ONIG_FEATURES=
 
@@ -49,4 +52,3 @@ matrix:
     - rust: nightly
       os: osx
       env: ONIG_FEATURES=std-pattern
-    

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -111,6 +111,7 @@ install:
 - cmd: git submodule update --init
 - ps: .\appveyor_rust_install.ps1
 - set PATH=C:\msys64\mingw%MSYS_BITS%\bin;C:\msys64\usr\bin;%PATH%
+- '"C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\SetEnv.cmd" /x64'
 
 # Alternative install command for simple configurations without build matrix (uncomment line and
 # comment above line):

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -6,6 +6,9 @@
 # Rust needs at least Visual Studio 2013 Appveyor OS for MSVC targets.
 os: Visual Studio 2015
 
+# Don't build pull requests twice
+skip_branch_with_pr: true
+
 ## Build Matrix ##
 
 # This configuration will setup a build for each channel & target combination (12 windows

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -30,12 +30,12 @@ environment:
 
 ### MSVC Toolchains ###
 
-  # # Stable 64-bit MSVC
-  #   - channel: stable
-  #     target: x86_64-pc-windows-msvc
-  # # Stable 32-bit MSVC
-  #   - channel: stable
-  #     target: i686-pc-windows-msvc
+  # Stable 64-bit MSVC
+    - channel: stable
+      target: x86_64-pc-windows-msvc
+  # Stable 32-bit MSVC
+    - channel: stable
+      target: i686-pc-windows-msvc
   # # Beta 64-bit MSVC
   #   - channel: beta
   #     target: x86_64-pc-windows-msvc

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -33,9 +33,13 @@ environment:
   # Stable 64-bit MSVC
     - channel: stable
       target: x86_64-pc-windows-msvc
+      TOOLCHAIN: msvc
+      PLATFORM: x86_64
   # Stable 32-bit MSVC
     - channel: stable
       target: i686-pc-windows-msvc
+      TOOLCHAIN: msvc
+      PLATFORM: i686
   # # Beta 64-bit MSVC
   #   - channel: beta
   #     target: x86_64-pc-windows-msvc
@@ -54,26 +58,32 @@ environment:
   # Stable 64-bit GNU
     - channel: stable
       target: x86_64-pc-windows-gnu
+      TOOLCHAIN: msys
       MSYS_BITS: 64
   # Stable 32-bit GNU
     - channel: stable
       target: i686-pc-windows-gnu
+      TOOLCHAIN: msys
       MSYS_BITS: 32
   # Beta 64-bit GNU
     - channel: beta
       target: x86_64-pc-windows-gnu
+      TOOLCHAIN: msys
       MSYS_BITS: 64
   # Beta 32-bit GNU
     - channel: beta
       target: i686-pc-windows-gnu
+      TOOLCHAIN: msys
       MSYS_BITS: 32
   # Nightly 64-bit GNU
     - channel: nightly
       target: x86_64-pc-windows-gnu
+      TOOLCHAIN: msys
       MSYS_BITS: 64
   # Nightly 32-bit GNU
     - channel: nightly
       target: i686-pc-windows-gnu
+      TOOLCHAIN: msys
       MSYS_BITS: 32
 
 ### Allowed failures ###
@@ -110,12 +120,10 @@ matrix:
 install:
 - cmd: git submodule update --init
 - ps: .\appveyor_rust_install.ps1
-- set PATH=C:\msys64\mingw%MSYS_BITS%\bin;C:\msys64\usr\bin;%PATH%
-- '"C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\SetEnv.cmd" /x64'
-
-# Alternative install command for simple configurations without build matrix (uncomment line and
-# comment above line):
-#- ps: .\appveyor_rust_install.ps1 -channel stable -target x86_64-pc-windows-msvc
+- if "%TOOLCHAIN%" == "msvc" if "%PLATFORM%" == "i686" call "%VS140COMNTOOLS%\..\..\VC\vcvarsall.bat"
+- if "%TOOLCHAIN%" == "msvc" if "%PLATFORM%" == "x86_64" "C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\SetEnv.cmd" /x64
+- if "%TOOLCHAIN%" == "msvc" if "%PLATFORM%" == "x86_64" call "%VS140COMNTOOLS%\..\..\VC\vcvarsall.bat" x86_amd64
+- if "%TOOLCHAIN%" == "msys" set PATH=C:\msys64\mingw%MSYS_BITS%\bin;C:\msys64\usr\bin;%PATH%
 
 ## Build Script ##
 

--- a/onig_sys/Cargo.toml
+++ b/onig_sys/Cargo.toml
@@ -7,7 +7,7 @@ links = "onig"
 
 description = """The Onig Sys crate contains raw rust bindings to the
 oniguruma library. This crate exposes a set of unsafe functions which
-can then be used by other craetes to create safe wrappers around
+can then be used by other crates to create safe wrappers around
 Oniguruma.
 
 You probably don't want to link to this crate directly."""
@@ -25,4 +25,6 @@ libc = "0.2"
 
 [build-dependencies]
 pkg-config = "0.3"
-cmake = "0.1"
+
+[target.'cfg(not(target_env="msvc"))'.build-dependencies.cmake]
+version = "0.1"

--- a/onig_sys/build.rs
+++ b/onig_sys/build.rs
@@ -1,12 +1,22 @@
 extern crate pkg_config;
+
+#[cfg(not(target_env="msvc"))]
 extern crate cmake;
 
 use std::env;
 
-fn compile_with_cmake() {
-    use cmake::Config;
+fn rustc_link_type(static_link: bool) -> &'static str {
+    if static_link {
+        "static"
+    } else {
+        "dylib"
+    }
+}
 
-    let static_link = env::var("CARGO_FEATURE_STATIC_ONIG").is_ok();
+
+#[cfg(not(target_env="msvc"))]
+fn compile(static_link: bool) {
+    use cmake::Config;
 
     // Builds the project in the directory located in `oniguruma`, installing it
     // into $OUT_DIR
@@ -19,21 +29,44 @@ fn compile_with_cmake() {
               }
               .build();
 
-    let link_type = if static_link {
-        "static"
-    } else {
-        "dylib"
-    };
-
     println!("cargo:rustc-link-search=native={}",
              dst.join("build").display());
-    println!("cargo:rustc-link-lib={}=onig", link_type);
+    println!("cargo:rustc-link-lib={}=onig", rustc_link_type(static_link));
+}
+
+#[cfg(target_env="msvc")]
+pub fn compile(static_link: bool) {
+    use std::process::Command;
+
+    // TODO: does this need to be dynamically looked up?
+    let onig_dir = env::current_dir().unwrap().join("oniguruma");
+    let build_dir = onig_dir.join("src");
+    let lib_name = if static_link { "onig_s" } else { "onig" };
+    
+    let bitness = if cfg!(target_pointer_width="64") { "64" } else { "32" };
+
+    let r = Command::new("cmd")
+        .args(&["/c", &format!("make_win{}.bat", bitness)])
+        .current_dir(&onig_dir)
+        .output()
+        .expect("error running build");
+
+    if !r.status.success() {
+        let err = String::from_utf8_lossy(&r.stderr);
+        let out = String::from_utf8_lossy(&r.stdout);
+        panic!("Build error:\nSTDERR:{}\nSTDOUT:{}", err, out);
+    }
+
+    println!("cargo:rustc-link-search=native={}", build_dir.display());
+    println!("cargo:rustc-link-lib={}={}", rustc_link_type(static_link), lib_name);
 }
 
 pub fn main() {
     if let Ok(_) = pkg_config::find_library("oniguruma") {
         return;
     }
+    
+    let static_link = env::var("CARGO_FEATURE_STATIC_ONIG").is_ok();
 
-    compile_with_cmake();
+    compile(static_link);
 }

--- a/onig_sys/build.rs
+++ b/onig_sys/build.rs
@@ -38,13 +38,13 @@ fn compile(static_link: bool) {
 pub fn compile(static_link: bool) {
     use std::process::Command;
 
-    // TODO: does this need to be dynamically looked up?
     let onig_dir = env::current_dir().unwrap().join("oniguruma");
     let build_dir = onig_dir.join("src");
     let lib_name = if static_link { "onig_s" } else { "onig" };
     
     let bitness = if cfg!(target_pointer_width="64") { "64" } else { "32" };
 
+    // Execute the oniguruma NMAKE command for the chosen architecture.
     let r = Command::new("cmd")
         .args(&["/c", &format!("make_win{}.bat", bitness)])
         .current_dir(&onig_dir)

--- a/src/names.rs
+++ b/src/names.rs
@@ -3,10 +3,7 @@ use std::iter::Iterator;
 use std::ptr::null;
 use std::str::from_utf8_unchecked;
 use std::slice::from_raw_parts;
-use libc::{c_int, c_uint, c_ulong, c_void, c_uchar};
-
-#[cfg(windows)]
-use libc::uintptr_t;
+use libc::{c_int, c_uint, c_void, c_uchar};
 
 use onig_sys::{self, OnigUChar, OnigRegex};
 
@@ -74,10 +71,10 @@ struct NameEntry {
 }
 
 #[cfg(windows)]
-type StDataT = uintptr_t;
+type StDataT = ::libc::uintptr_t;
 
 #[cfg(not(windows))]
-type StDataT = c_ulong;
+type StDataT = ::libc::c_ulong;
 
 #[repr(C)]
 #[derive(Debug)]

